editor/src/core/property-controls/property-controls-local.ts

```diff
@@ -174,49 +175,97 @@ export function parseComponentToRegister(value: unknown): ParseResult<ComponentT
   )
 }
 
-export const parseComponents: (
+const parseComponents: (
   value: unknown,
 ) => ParseResult<{ [componentName: string]: ComponentToRegister }> = parseObject(
   parseComponentToRegister,
 )
 
+function parseAndPrepareComponents(
+  workers: UtopiaTsWorkers,
+  moduleNameOrPath: string,
+  unparsedComponents: unknown,
+): Either<string, PreparedComponentDescriptorsForRegistering> {
+  const parsedComponents = parseComponents(unparsedComponents)
+
+  return bimapEither(
+    (parseError) => {
+      const errorDetails = getParseErrorDetails(parseError)
+      return `registerModule second param (components): ${errorDetails.description} [${errorDetails.path}]`
+    },
+    (components: { [componentName: string]: ComponentToRegister }) => {
+      return prepareComponentDescriptorsForRegistering(workers, moduleNameOrPath, components)
+    },
+    parsedComponents,
+  )
+}
+
+type PartiallyAppliedParseAndPrepareComponents = (
```

Comment:
could you put Fn in the name of the type? I think that'd make the type of cachedParseAndPrepareComponentsMap clearer